src/components/Hoverable/index.tsx

```diff
@@ -27,21 +26,18 @@ function mapChildren(children, callbackParam) {
 
 /**
  * Assigns a ref to an element, either by setting the current property of the ref object or by calling the ref function
- * @param {Object|Function} ref - The ref object or function.
- * @param {HTMLElement} el - The element to assign the ref to.
+ * @param ref The ref object or function.
+ * @param element The element to assign the ref to.
  */
-function assignRef(ref, el) {
+function assignRef(ref: ((instance: HTMLElement | null) => void) | MutableRefObject<HTMLElement | null>, element: HTMLElement) {
     if (!ref) {
         return;
     }
-
-    if (_.has(ref, 'current')) {
+    if (typeof ref === 'function') {
+        ref(element);
+    } else if (ref?.current) {
```

Comment:
This was overlooked. `if (_.has(ref, 'current')) {` is equivalent to:
```suggestion
    } else if (ref.current !== undefined) {
```